File: backend/app/models/youtube.py
from pydantic import BaseModel


class YTVideo(BaseModel):
    video_id: str
    title: str
    url: str
    views: int
    category: str


class YTChannelPreview(BaseModel):
    channel_title: str
    channel_url: str
    total_videos: int
    categories: dict[str, int]
    videos: list[YTVideo]
    all_videos: list[YTVideo]

File: next-frontend/lib/types/youtube.ts
export interface YTVideo {
  video_id: string;
  title: string;
  url: string;
  views: number;
  category: string;
  is_transcribed: boolean;
}

export interface YTChannelPreview {
  channel_title: string;
  channel_url: string;
  total_videos: number;
  categories: Record<string, number>;
  videos: YTVideo[];
}
